modules/Story/Share/Share.tsx

```diff
@@ -0,0 +1,145 @@
+'use client';
+
+import type { Story } from '@prezly/sdk';
+import { translations, useIntl } from '@prezly/theme-kit-nextjs/index';
+import classNames from 'classnames';
+
+import { Button, ButtonLink } from '@/components/Button';
+import { Divider } from '@/components/Divider';
+import { SocialShare } from '@/components/SocialShare';
+import { IconFileDown, IconFolderDown, IconLink, IconText } from '@/icons';
+import type { StoryActions } from 'theme-settings';
+
+import type { SharingOptions } from '../type';
+
+import { ButtonWithSuccessTooltip } from './ButtonWithSuccessTooltip';
+import { getAssetsArchiveDownloadUrl } from './utils/getAssetsArchiveDownloadUrl';
+import { getStoryPdfUrl } from './utils/getStoryPdfUrl';
+
+import styles from './Share.module.scss';
+
+interface Props {
+    actions?: StoryActions;
+    sharingOptions: SharingOptions;
+    thumbnailUrl?: string;
+    title: string;
+    url: string;
+    uploadcareAssetsGroupUuid: Story.ExtraFields['uploadcare_assets_group_uuid'];
+    slug: Story['slug'];
+    uuid: Story['uuid'];
+}
+
+export function Share({
+    actions,
+    uploadcareAssetsGroupUuid,
+    sharingOptions,
+    slug,
+    title,
+    thumbnailUrl,
+    url,
+    uuid,
+}: Props) {
+    const { formatMessage } = useIntl();
+    const socialNetworks = sharingOptions.sharing_actions;
+    const socialShareButtonsCount = socialNetworks.length;
+    const actionsButtonsCount = [
+        actions?.show_copy_content,
+        actions?.show_copy_url,
+        actions?.show_download_assets,
+        actions?.show_download_pdf,
+    ].filter(Boolean).length;
+    const assetsUrl = uploadcareAssetsGroupUuid
+        ? getAssetsArchiveDownloadUrl(uploadcareAssetsGroupUuid, slug)
+        : undefined;
+
+    function handleCopyLink() {
+        window.navigator.clipboard.writeText(url);
+    }
+
+    async function handleCopyText() {
+        const { copyStoryText } = await import('./utils/copyStoryText');
+        copyStoryText();
+    }
+
+    async function handlePdfDownload() {
+        const pdfUrl = await getStoryPdfUrl(uuid);
+        const link = document.createElement('a');
+        link.setAttribute('href', pdfUrl);
+        link.setAttribute('download', `${title}.pdf`);
+        document.body.appendChild(link);
+        link.click();
+        document.body.removeChild(link);
+    }
+
+    return (
+        <>
+            <Divider />
+            <div>
+                <h2>Share</h2>
+
```

Comment:
This'll be out of place on non-English sites. We'll probably have to add translations.

Reply:
I would need some assistance with that